src/atama/receiver/node/receiver_node.cpp

```diff
@@ -0,0 +1,112 @@
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "atama/receiver/node/receiver_node.hpp"
+#include "kansei_interfaces/msg/orientation.hpp"
+#include "ninshiki_interfaces/msg/detected_object.hpp"
+#include "ninshiki_interfaces/msg/detected_objects.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tachimawari_interfaces/srv/get_joints.hpp"
+
+using namespace std::chrono_literals;
+
+namespace atama::receiver
+{
+
+ReceiverNode::ReceiverNode(rclcpp::Node::SharedPtr node, std::shared_ptr<atama::head::Head> head)
+: node(node), head(head)
+{
+  using kansei_interfaces::msg::Orientation;
+  using ninshiki_interfaces::msg::DetectedObject;
+  using ninshiki_interfaces::msg::DetectedObjects;
+  using tachimawari_interfaces::srv::GetJoints;
+
+  get_joints_client = node->create_client<GetJoints>(
+    "/joint/get_joints");
+
+  get_orientation_subsciber = node->create_subscription<Orientation>(
+    "measurement/orientation", 10,
+    [this](const Orientation::SharedPtr message) {
+      // this->head->set_yaw(message->orientation[2]);
+    }
+  );
+
+  get_detection_result_subsciber =
+    node->create_subscription<DetectedObjects>(
+    "ninshiki_py/detection", 10,
+    [this](const DetectedObjects::SharedPtr message) {
+      std::vector<DetectedObject> temp_detection_result;
+
+      for (const auto & detected_object : message->detected_objects) {
+        temp_detection_result.push_back(detected_object);
+      }
+
+      this->head->detection_result = temp_detection_result;
+    }
+    );
+}
+
+bool ReceiverNode::get_joints_data()
+{
+  while (!get_joints_client->wait_for_service(1s)) {
+    if (rclcpp::ok()) {
+      // service not available, waiting again...
+    } else {
+      // Interrupted while waiting for the service. Exiting.
+      return false;
+    }
+  }
+  auto result = get_joints_client->async_send_request(
+    std::make_shared<tachimawari_interfaces::srv::GetJoints::Request>());
+
+  if (rclcpp::spin_until_future_complete(node, result) ==
+    rclcpp::FutureReturnCode::SUCCESS)
+  {
+    std::vector<tachimawari::joint::Joint> temp_joints;
+
+    for (const auto & joint : result.get()->joints) {
+      for (std::string id : {"neck_yaw", "neck_pitch"}) {
+        uint8_t joint_id = tachimawari::joint::JointId::by_name.find(id)->second;
+
+        if (joint.id == joint_id) {
+          temp_joints.push_back(
+            tachimawari::joint::Joint(joint.id, joint.position));
+        }
+      }
+    }
+
+    head->set_joints(temp_joints);
```

Comment:
i suggest to do this instead:
- create a `std::set` from list of `{"neck_yaw", "nectk_pitch"}`, find the id and push to that `std::set`
- for each `result.get()->joints`, check whether `joint.id` is in the `std::set`, push to `temp_joints` if it is

Reply:
done